pkg/infra/iac2/iac_templates.go

```diff
@@ -96,7 +99,12 @@ func doQuery(c *sitter.Node, q string) query.NextMatchFunc {
 
 // parameterizeArgs turns "args.foo" into {{.Foo}}. It's very simplistic and just works off regex
 func parameterizeArgs(contents string) string {
-	return parameterizeArgsRegex.ReplaceAllString(contents, `{{$1}}`)
+	// If the source has "{args.Foo}", then just turning "args.Foo" -> "{{.Foo}}" would result in "{{{.Foo}}}", which is
+	// invalid go-template. So, we first turn "{args." into "{{`{`}}args.", which will eventually result in
+	// "{{`{`}}{{.Foo}}" — which, while ugly, will result in the correct template execution.
+	contents = curlyEscapes.ReplaceAllString(contents, "{{`$1`}}args")
```

Comment:
Just to make sure, this would be valid right? `{{'{'}}{{.Foo}}}` (the ending triple `}}}`)

Reply:
Yep, that's valid (and is in fact what we generate).